App.js

```diff
@@ -116,37 +237,77 @@ export default function App() {
         workOdd = false;
       }
       if (workOdd === true && prevGameState.even === false) {
-        workMessage = 'Roll again';
-        workOptMessage = 'ODD selected';
+        workMessage = 'Roll again (ODD only)';
       }
       if (workOdd === false && prevGameState.even === true) {
-        workMessage = 'Roll again';
-        workOptMessage = 'EVEN selected';
+        workMessage = 'Roll again (EVEN only)';
       }
       return {
         roll: prevGameState.roll,
         even: prevGameState.even,
         odd: workOdd,
         position1: prevGameState.position1,
+        position2: prevGameState.position2,
         message: workMessage,
-        optMessage: workOptMessage,
+        optMessage: prevGameState.optMessage,
         score: prevGameState.score,
         board: prevGameState.board,
         endOfGame: prevGameState.endOfGame,
         about: prevGameState.about,
+        numberOfPlayers: prevGameState.numberOfPlayers,
+        whoseTurn: prevGameState.whoseTurn,
+        turnTotals: prevGameState.turnTotals,
       };
     });
   }, []);
 
+  // calculate whose turn it is
+  const calculateTurn = (workTurnTotals,P1,P2) => {
```

Comment:
Can you add a better description of what this function is, and what the parameters mean?

Reply:
Done.